### DISCUSSION TITLE
Migration JS delete initial 'users' collection

### DISCUSSION THREAD

**Discussion by Sazzels** [pocketbase/pocketbase#3508] [Q&A] [answered]
hi, 

it's probably super obvious but i'm trying to delete the initial collection 'users' using migrations. Unfortunately I don't seem to succeed. 

I created a default pocketbase docker image without making any changes and deleted the users collection from the webinterface. Because `migrations` are enabled by default, i could have a look in the corresponding folder within the container. There i found out that the collections `id` is `_pb_users_auth_`.

```bash
# Dockerfile
...
# uncomment to copy the local pb_migrations dir into the image
COPY ./pb_migrations /pb/pb_migrations
...
```

```javascript
// pb_migrations/10_delete_initial_users.js
migrate((db) => {
  const dao = new Dao(db);
  const collection = dao.findCollectionByNameOrId("_pb_users_auth_");

  return dao.deleteCollection(collection);
});
```

Unfortunately with this migration file I always get the following error when starting the container

```bash
[timestamp] Failed to apply migration 10_delete_initial_users.js: SQL logic error: no such table: _collections (1)
```

> the reason for all this is, I would like to create a default set of collections when creating the image. This way I could keep the structure of the collections in Git and edit them there. Then build new images via pipeline (CI/CD).

**Comment by ganigeorgiev** [comment, marked as answer]
Migrations are applied in their filename order.
In your case `10_delete_initial_users.js` will be executed before the system migrations (_the system migrations will eventually change in the future to have a negative/zero-ed value to avoid the conflict with user migrations_).

For now, it is recommended to use a timestamp value as prefix for your migration file, for example: `1697193097_delete_initial_users.js`.

_You can also use the builtin `./pocketbase migrate create 'delete initial users'` command that will generate a blank template migration file for you._

**Reply by Sazzels** [reply]
this worked flawless, thank you very much!
